docs/README.md

```diff
@@ -62,6 +62,52 @@ Alternatively, you can use the `scripts/build_linux.sh` script to perform the sa
 ./scripts/build_linux.sh
 ```
 
+### Code Formatting and Pre-commit Hooks
+
+This project uses [pre-commit](https://pre-commit.com) to ensure code quality and consistent formatting.
+
+To set it up:
+
+1. Install `pre-commit` if you don't have it:
+
+   ```bash
+   pip install pre-commit
```

Comment:
it would be nice to mention the `venv` . Installing `pre-commit` globally may not be the best idea.